core/ipc/rpc/kafka/src/main/java/org/opennms/core/ipc/rpc/kafka/KafkaRpcClientFactory.java

```diff
@@ -181,6 +185,9 @@
                         expirationTime, loggingContext, request.getLocation(), span);
                 delayQueue.offer(responseHandler);
                 rpcResponseMap.put(rpcId, responseHandler);
+                if(kafkaConsumerRunner.isClosed() && !kafkaConsumerExecutor.isShutdown()) {
```

Comment:
I think it would be cleaner if we did not rely on the thread to set the `closed` flag when it stopped and use some other mechanism instead. The `closed` flag is now being used a both a request to stop the thread, and one to track the state of the thread.

Instead of calling `execute` on the executor, we can use `submit` and query the resulting future with `isDone` to see if the thread is truly stopped or not.

Reply:
Might be missing something here... but is this restart logic actually desired and is it even ever going to execute? The only thing that stops the consumer right now is this bean being torn down. If this bean has been torn down the executor should be shut down/shutting down and it wouldn't make sense to start the Kafka consumer up again.

The consumer is also swallowing all `Throwable`s so it isn't going to stop due to an exception. So we should be able to safely assume once it is started, it will remain so until the bean is torn down.

Reply:
You are right. Went ahead and removed logic to restart thread.